File: 485-max-consecutive-ones/max-consecutive-ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {

        if(nums.empty()) return 0;

        int count = 0;
        int maxones = 0;
        for(int i = 0; i < nums.size(); i++) {
           if(nums[i] == 1){
                count++;
                maxones = max(maxones,count);
           }
           if(nums[i] != 1){
                count = 0;
           }
        }
    return maxones;
    }
};

File: 485-max-consecutive-ones/max-consecutive-ones.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int max = 0;
        int currentcount = 0;

        for (int i = 0; i < nums.length; i++) {
            if (nums[i] == 1) {
                currentcount++;
            } else {
                currentcount = 0;
            }
            max = Math.max(max, currentcount);
        }
        return max;
    }
}